src/Client.php

```diff
@@ -394,15 +394,29 @@ public function send($method, $payload, $path, $params = [], array $headers = []
 
         $uri = urldecode($location->getPath());
 
-        // Rewind stream
-        // see: https://github.com/guzzle/psr7/issues/38
-        $response->getBody()->rewind();
-
         // Unserialize response body
-        $content = json_decode($response->getBody()->getContents(), true);
+        $content = json_decode((string) $response->getBody(), true) ?: [];
 
         // Build expected resource
-        $resource = $this->factory->create($uri, $content);
+        $resource = $this->factory->create($uri, []);
```

Comment:
with this change the above variable `$content` gets used only in the first part of the following `if` so the above unserialize can be moved inside the if ?